[pickers] Declaration files don't type-check with TypeScript 4.8

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

The declaration files currently have a error with TypeScript 4.8:

```
ERROR in node_modules/@mui/x-date-pickers/internals/models/muiPickersAdapter.d.ts:2:55
TS2344: Type 'TDate' does not satisfy the constraint 'ExtendableDateType'.
    1 | import { IUtils } from '@date-io/core/IUtils';
  > 2 | export declare type MuiPickersAdapter<TDate> = IUtils<TDate>;
      |                                                       ^^^^^
    3 | export declare type MuiDateSectionName = 'day' | 'month' | 'year' | 'hour' | 'minute' | 'second' | 'am-pm';
    4 | export declare type MuiFormatTokenMap = {
    5 |     [formatToken: string]: MuiDateSectionName;
```

### Expected behavior 🤔

The declaration files should not have an error with TypeScript 4.8.

### Steps to reproduce 🕹

Create TypeScript 4.8 project with `@mui/x-date-pickers` with `skipLibCheck` turned off.

### Context 🔦

I noticed this was addressed in https://github.com/mui/mui-x/pull/5935, but wanted to make sure that the fix made it into v5 since [the release notes for the latest v5 release](https://github.com/mui/mui-x/releases/tag/v5.17.2) mentioned that this was the last scheduled regular release for v5.

### Your environment 🌎

N/A

### Order ID 💳 (optional)

_No response_

Thanks for the update, but I've just updated my project to use `"@mui/x-date-pickers": "^5.0.2"` and `"typescript": "^4.8.3"` but I'm still getting the exact same error:

```
> connect@1.0.0 typecheck
> tsc --noEmit

node_modules/@mui/x-date-pickers/internals/models/muiPickersAdapter.d.ts:2:55 - error TS2344: Type 'TDate' does not satisfy the constraint 'ExtendableDateType'.

2 export declare type MuiPickersAdapter<TDate> = IUtils<TDate>;
                                                        ~~~~~

  node_modules/@mui/x-date-pickers/internals/models/muiPickersAdapter.d.ts:2:39
    2 export declare type MuiPickersAdapter<TDate> = IUtils<TDate>;
                                            ~~~~~
    This type parameter might need an `extends ExtendableDateType` constraint.


Found 1 error in node_modules/@mui/x-date-pickers/internals/models/muiPickersAdapter.d.ts:2
```

This is currently blocking our build, so I'm going to have to turn on skipLibCheck for now...

--- a/packages/x-date-pickers/src/internals/models/muiPickersAdapter.d.ts
+++ b/packages/x-date-pickers/src/internals/models/muiPickersAdapter.d.ts
@@ -1,7 +1,7 @@
 import { IUtils } from '@date-io/core/IUtils';
 
-// @ts-ignore TDate in our codebase does not have the `ExtendableDateType` constraint.
 // TODO: Maybe we should add the same constraint.
+// @ts-expect-error TDate in our codebase does not have the `ExtendableDateType` constraint.
 export type MuiPickersAdapter<TDate> = IUtils<TDate>;
 
 export type MuiDateSectionName = 'day' | 'month' | 'year' | 'hour' | 'minute' | 'second' | 'am-pm';

